vulnerabilities/pipelines/remove_duplicate_advisories.py

```diff
@@ -0,0 +1,83 @@
+#
+# Copyright (c) nexB Inc. and others. All rights reserved.
+# VulnerableCode is a trademark of nexB Inc.
+# SPDX-License-Identifier: Apache-2.0
+# See http://www.apache.org/licenses/LICENSE-2.0 for the license text.
+# See https://github.com/aboutcode-org/vulnerablecode for support or download.
+# See https://aboutcode.org for more information about nexB OSS projects.
+#
+
+import logging
+from itertools import groupby
+
+from aboutcode.pipeline import LoopProgress
+from django.db.models import Count
+from django.db.models import Q
+
+from vulnerabilities.models import Advisory
+from vulnerabilities.pipelines import VulnerableCodePipeline
+from vulnerabilities.utils import compute_content_id
+
+
+class RemoveDuplicateAdvisoriesPipeline(VulnerableCodePipeline):
+    """Pipeline to remove duplicate advisories based on their content."""
+
+    pipeline_id = "remove_duplicate_advisories"
+
+    @classmethod
+    def steps(cls):
+        return (
+            cls.recompute_content_ids,
+            cls.remove_duplicates,
+        )
+
+    def remove_duplicates(self):
+        """
+        Find advisories with the same content and keep only the latest one.
+        """
+
+        duplicated_advisories = groupby(
+            Advisory.objects.order_by("unique_content_id").all().paginated(),
+            key=lambda x: x.unique_content_id,
+        )
```

Comment:
I honestly doubt that this will work smoothly in production, ordering 118 million advisories using `unique_content_id`, which is an unindexed field is not good idea.

https://github.com/aboutcode-org/vulnerablecode/pull/1766#discussion_r1946776376 would be much more practical approach since `id` is an autogenerated primary key field and hence it's already indexed, and we can use this to select the latest or oldest advisory.

Reply:
db_index is in models for that! And since we believe there are lots of dupes. I believe index to work fine

Reply:
We're adding this `db_index` now. I am not sure how long the index creation itself is going to take.